### ISSUE TITLE
Again detected as Malware

### ISSUE THREAD

**Issue by neoOpus** [aloneguid/bt#27]
After reinstalling Windows and configuring Bluetooth, MS Defender removed it after identifying it as malware (I didn't have time to add it to a whitelist)

Please figure out why it's happening and get it signed now. Even if it's free, some people might not like using it because it might scare them or be taken off their computers.

![Screenshot 2023-08-11 102854](https://github.com/aloneguid/bt/assets/1388336/df9e5a8a-6c4b-4718-8c42-148a88e35816)
![image](https://github.com/aloneguid/bt/assets/1388336/b0b1cef7-d215-45e9-8b6e-a1d5379e6049)

**Comment by neoOpus** [comment]
I am unable to download it unless, of course, if I disable MS Defender

![image](https://github.com/aloneguid/bt/assets/1388336/d17b2475-3111-49a5-9e7c-23afcf8f4b7c)

**Comment by aloneguid** [comment]
I can't afford a signing certificate so it's bit going to happen. You are free to validate it's not dangerous as source code and build pipelines are completely open and transparent.

**Comment by corvus2606** [comment]
> I can't afford a signing certificate so it's not going to happen. You are free to validate it's not dangerous as source code and build pipelines are completely open and transparent.

What is the cost of a signing certificate?

**Comment by aloneguid** [comment]
It's [about $1.5k for 3 years](https://order.digicert.com/step1/code_signing). Could be less if you shop around. But that won't solve false AV issues, you can still be banned and certificate revoked for no reason. I think realistically one needs a legal team to deal with AV false claims which I apparently don't have. I'd recommend having a read:

- https://weblog.west-wind.com/posts/2016/Oct/05/Dealing-with-AntiVirus-False-Positives
- https://www.autohotkey.com/boards/viewtopic.php?t=87322
- https://steamcommunity.com/app/779590/discussions/0/2572002906839928114/
- https://help.steampowered.com/en/faqs/view/5F3D-1477-AFF9-C4F3
- https://www.linkedin.com/pulse/kaspersky-reasons-false-positives-amirabbas-mahdavi
- https://www.gdatasoftware.com/blog/2022/06/37445-malware-detection-is-hard
- https://medium.com/@airflow.matt/globalsign-will-revoke-your-codesign-certificate-no-questions-asked-f6ac2bca02c5

And by the way, the last BT version (3.5.0) has only a single AV's claim out of 90, unlike 29 out of 90 for version 3.4.0, so it's totally random trash. I've myself became very pessimistic about usefulness of AV software in general after dealing with this.

**Comment by paz** [comment]
Kaspersky and Sophos both left BT undetected for me, seems it might be a Microsoft specific issue.

**Comment by aloneguid** [comment]
it's totally random and changes daily ;)

**Comment by jnv** [comment]
Same thing here, BT 3.5.2 was flagged by Microsoft Defender as PUA. It's possible to send files to Microsoft for further analysis: https://www.microsoft.com/en-us/wdsi/filesubmission/ – I urge you to do it if you are affected.

**Comment by aloneguid** [comment]
> Same thing here, BT 3.5.2 was flagged by Microsoft Defender as PUA. It's possible to send files to Microsoft for further analysis: https://www.microsoft.com/en-us/wdsi/filesubmission/ – I urge you to do it if you are affected.

I have used this before, and just submitting for latest version as "incorrectly identified as malware". Will let you know on progress:
![image](https://github.com/aloneguid/bt/assets/3155189/b2657335-a79d-4076-9ee7-084aac99c5da)

**Comment by aloneguid** [comment]
Analysis on above is still pending but some detections have already cleared out.

**Comment by neoOpus** [comment]
It keeps getting deleted even when excluded from scans... I have to reinstall it every few days.

**Comment by CityguyUSA** [comment]
There are 2 different .zip files.  A pdb version which downloads fine and non-pdb version that doesn't.  What's the difference between the 2?

**Comment by aloneguid** [comment]
> There are 2 different .zip files. A pdb version which downloads fine and non-pdb version that doesn't. What's the difference between the 2?

.pdb version is debug symbols to investigate crashes, you don't need that.

**Comment by aloneguid** [comment]
> It keeps getting deleted even when excluded from scans... I have to reinstall it every few days.

You can permanently allow the "threat" until MS investigates. There are instructions available [here](https://pureinfotech.com/allow-blocked-file-app-microsoft-defender-antivirus/#:~:text=1%20To%20allow%20an%20item%20that%20has%20been,recommended%20to%20read%20the%20details%20and%20understand%20the).

**Comment by aloneguid** [comment]
Windows Defender should now be fine, just got analysis results from Microsoft:

![image](https://github.com/aloneguid/bt/assets/3155189/bb09fc2b-3853-4e03-af64-d01fb01adffb)

**Comment by aloneguid** [comment]
Also VirusTotal before and after (Microsoft AV is OK now). Hopefully others will follow the suit.

![image](https://github.com/aloneguid/bt/assets/3155189/eaaa6f97-2814-47d6-8b5f-4447078b76a1)

![image](https://github.com/aloneguid/bt/assets/3155189/3d430ec7-17e2-466f-b6b1-9b38006a0213)

**Comment by neoOpus** [comment]
> > It keeps getting deleted even when excluded from scans... I have to reinstall it every few days.
> 
> You can permanently allow the "threat" until MS investigates. There are instructions available [here](https://pureinfotech.com/allow-blocked-file-app-microsoft-defender-antivirus/#:~:text=1%20To%20allow%20an%20item%20that%20has%20been,recommended%20to%20read%20the%20details%20and%20understand%20the).

I have been doing that since the start, but it doesn't stick. That's why I notified you that currently it is allowed and working properly, but it crashes when trying to find updates... I am simply informing you of this, but it is alright if you are unable to resolve these.

**Comment by aloneguid** [comment]
@neoOpus thanks. Update checks are already fixed and will be out in v3.6. Defender does not block it anymore.

**Comment by aloneguid** [comment]
@jnv thanks for that, I also raised request with [ESET](https://intelix.sophos.com/) which has reclassified as clean.

**Comment by aloneguid** [comment]
@jnv I have raised Avast issue separately yesterday, and classification is cleared completely.

**Comment by aloneguid** [comment]
Also submitted a dispute to McAfee now.

**Comment by aloneguid** [comment]
And just for fun to [Malwarebytes](https://forums.malwarebytes.com/topic/302499-browser-tamer-false-positive/).

**Comment by neoOpus** [comment]
So far, it worked and I didn't have any issue :)

**Comment by jnv** [comment]
Unfortunately we're back to [24](https://www.virustotal.com/gui/file/7273f03b70a07ab0e1cf96aa4702587ea850b72efbdeef4690bf44bb3edd295b/details) false positives for the v3.6.2 installer. Today even Microsoft Defender took down my locally compiled version.

**Comment by aloneguid** [comment]
Microsoft seems to be happy, but others are not. It will help in long term to vote on VirusTotal community webpage: https://www.virustotal.com/gui/file/7273f03b70a07ab0e1cf96aa4702587ea850b72efbdeef4690bf44bb3edd295b/community

**Comment by aloneguid** [comment]
AVG and Avast were great help in whitelistimg 3.6.2, we are -2 now.

**Comment by aloneguid** [comment]
14/61 today!

![image](https://github.com/aloneguid/bt/assets/3155189/bfb841a2-96d5-491a-a3da-296be207cea4)

**Comment by cheTesta** [comment]
![image](https://github.com/aloneguid/bt/assets/24191291/3d351a94-73be-4673-8181-48d7ef948257)
18 on the .zip version

**Comment by aloneguid** [comment]
Down to 11 today:

![image](https://github.com/aloneguid/bt/assets/3155189/1033f9fc-63a1-4254-b373-1371f55ba309)

**Comment by aloneguid** [comment]
10 today.

**Comment by Ultrafeel** [comment]
https://www.virustotal.com/gui/file/9a6a86a90c1c68423465a4b800f4f2941a92e1f82ed9ea0f4dfb58a641932cef/details

**Comment by eiqnepm** [comment]
It seems Microsoft doesn't like the latest version as it has been automatically removed from my PC by Windows Defender.

**Comment by aloneguid** [comment]
Same here, looks like Microsoft needs a ticket for every new version to allowlist it again.

**Comment by aloneguid** [comment]
Also this is odd, because VirusTotal only reports 4 hits from "bad" AVs

![image](https://github.com/aloneguid/bt/assets/3155189/78e70d2d-784f-40dd-8cac-37fc41d862cf)

**Comment by aloneguid** [comment]
Yeah, but it would be also super slow and at least x100 bigger in size of downloads and ram.

**Comment by mahoromax** [comment]
> It seems Microsoft doesn't like the latest version as it has been automatically removed from my PC by Windows Defender.

Not only the latest, also 1 or 2 previous - it broke, I installed newer one (deactivated the antivir) but at some point it reactivated... It actually breaks opening links if BT is set as default handle for hyperlinks. Typical Microsoft -.-

Can we whitelist it manually? Or do we need to do that for every version as well?

**Comment by Ultrafeel** [comment]
From what I understand from this talk about Windows - BlueHat IL 2023 - David Weston - Default Security
 https://www.youtube.com/watch?v=8T6ClX-y2AE : maybe turning in UWP and converting msi to MSIX can make help against AV 🤓. One of the issues was mentioned is "over privileged apps".

**Comment by dmocha** [comment]
I downloaded version 4.0.2 today
Viirus total shows a result of 21/73
https://www.virustotal.com/gui/file/68d876a6afb7d9bef7b1b1e1ebe32eea68ce7bd83f4c56c216cc25c0a90e3d4a
It's starting to make you think
:(

**Comment by aloneguid** [comment]
Please finish your thought ;)

**Comment by eiqnepm** [comment]
> I downloaded version 4.0.2 today Viirus total shows a result of 21/73 https://www.virustotal.com/gui/file/68d876a6afb7d9bef7b1b1e1ebe32eea68ce7bd83f4c56c216cc25c0a90e3d4a It's starting to make you think :(

Browser Tamer by design needs to intercept links to work, this sometimes sets off the real time behavioral analysis of AV engines, when triggered most AVs will decided if it's malware based on reputation and manual review, obviously Browser Tamer is a small project with not a lot of users in the grand scheme of things, so the reputation is going to be low. Microsoft has already whitelisted versions of Browser Tamer after being submitted for manual review.

Browser Tamer releases are built using GitHub Actions, and of course you can build it yourself too.

**Comment by dmocha** [comment]
> Please finish your thought ;)

I choose consciously and accept the possible risks, but others, seeing the scale, may give up at the very beginning

**Comment by aloneguid** [comment]
Latest build has 2 warnings instead of 21, so it's totally random

![image](https://github.com/user-attachments/assets/f7f7734e-b1f2-4166-9604-461d2600874e)

**Comment by aloneguid** [comment]
If anyone knows how to contact "Sangfor" (chinese company) to submit false positive report please let me know.

**Comment by dmocha** [comment]
Have you tried using the website: https://www.sangfor.com/about-us/contact-us
?
I'll add right away that I just looked it up on the Internet, I haven't used it myself

**Comment by aloneguid** [comment]
Thanks @dmocha, trying this one.

**Comment by aloneguid** [comment]
Unfortunately that didn't work - no response for a week or so. However, after a lot of comms with various AV providers I have managed to reduct VT to only 1 (Sangfor):

![image](https://github.com/user-attachments/assets/dbdfa03d-4dd9-44b4-87d4-268f6ebbed0f)

**Comment by neoOpus** [comment]
> Unfortunately that didn't work - no response for a week or so. However, after a lot of comms with various AV providers I have managed to reduct VT to only 1 (Sangfor):
> 
> ![image](https://private-user-images.githubusercontent.com/3155189/366384580-dbdfa03d-4dd9-44b4-87d4-268f6ebbed0f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwNjk5MTUsIm5iZiI6MTcyNjA2OTYxNSwicGF0aCI6Ii8zMTU1MTg5LzM2NjM4NDU4MC1kYmRmYTAzZC00ZGQ5LTQ0YjQtODdkNC0yNjhmNmViYmVkMGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MTFUMTU0NjU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTcyNzRjMGRlYmMzNzlhOTMzYmQ3OGRhOWMwNDA4YTdjZDM2MmNlYjZlNjYzNWIxYmE3ZDVkMjU0NjNiZGVkMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.0sZTcEgwCZRAuZWMYdl7S4hWbbqOsvn2y6Or6ktghHE)

Maybe they will be more responsive if you contact them via their social media (https://x.com/SANGFOR)

**Comment by aloneguid** [comment]
@neoOpus seems like you have to be a premium user to contact them in the eX-twitter.

**Comment by neoOpus** [comment]
I tried but maybe it is just like you said... I will keep you updated if they reply

**Comment by aloneguid** [comment]
this is THE FIRST TIME we have [ZERO virus detections](https://www.virustotal.com/gui/file/3c57b955971518a88b140d1279073c932e9e095631558917d572dc7c1f2e7607?nocache=1)! I'm not sure what you did @neoOpus but it seems to have worked!

![image](https://github.com/user-attachments/assets/49831528-f2da-4a43-81f3-f477285a02a4)

**Comment by dmocha** [comment]
👍
Indeed, the bt-4.1.2.zip archive is "clean".
I just checked this archive

Now it would be useful to do the same with the bt.exe file

Because the file sent to VT indicates two threats:

![2024-10-03 20 56 48 www virustotal com 943cbbbd2956](https://github.com/user-attachments/assets/ab4aa41f-32a8-4b52-8345-66081058e236)
Link:
https://www.virustotal.com/gui/file/7908c83bdbf21dc49c9f8f8b644cec36f806d65b57792eb54faeb97c3d3c7cb4?nocache=1

**Comment by aloneguid** [comment]
@dmocha only Chinese AV companies seems to complain ;)

**Comment by aloneguid** [comment]
4.2.0

![image](https://github.com/user-attachments/assets/3f2ae657-f839-4aeb-9149-7b472f45c43d)

**Comment by dmocha** [comment]
Version 5.1.1.0
https://www.virustotal.com/gui/file/9d5b10958f0f679affcea744c91f6c7366dce3718bc81cd66bbac967fd4f5fc5

<img width="1356" height="500" alt="Image" src="https://github.com/user-attachments/assets/27a5c8fd-ad22-46e2-a450-f368627cb3ed" />

**Comment by dmocha** [comment]
The extracted file in version 5.2.2 shows two issues. As you mentioned earlier, we can skip Bkav, but the second one is Microsoft

<img width="1347" height="462" alt="Image" src="https://github.com/user-attachments/assets/909d2dfc-bf19-47dd-97d5-0f29391fb523" />

**Comment by trajano** [comment]
ESET just flagged and deleted the bt.exe file from scoop

> 2026-04-08 12:19:15 PM;Startup scanner;file;c:\users\trajano\scoop\apps\bt\current\bt.exe;a variant of Win64/Kryptik.GRS trojan;cleaned by deleting;

https://support.eset.com/en/kb3345-how-do-i-whitelist-my-software-with-eset

Downgrading to 5.5.6 gets around the issue

**Comment by dmocha** [comment]
I don’t know what happened, but something is not right.
A moment ago I checked the link to the ZIP archive on VirusTotal, and ten vendors flag it as a Trojan.
I also checked the bt.exe file on VirusTotal, and it was flagged as a Trojan by fourteen vendors.

![Image](https://github.com/user-attachments/assets/b0330984-2c88-4985-83b6-2199e5a2cbf0)
![Image](https://github.com/user-attachments/assets/e13f73fc-d681-49c2-a199-eb3f40522f58)

**Comment by aloneguid** [comment]
This just shows how broken the AV industry is and how useless AV software is in general. I will submit reports to clean this.

**Comment by aloneguid** [comment]
5.6.2 is [back to normal](https://www.virustotal.com/gui/file-analysis/ZDg1ODA0MTlmYWQ5NzgzNjNkNzFmOTYxMzRjM2QyNzc6MTc3NjI0MDAzMg==).

<img width="2045" height="765" alt="Image" src="https://github.com/user-attachments/assets/1563ee10-a3dc-4cf1-ae94-50b511eea17d" />

**Comment by Abeatbowl** [comment]
Hello, 
I've got the issue when trying to download [BrowserTamer-6.1.0-win64.msi](https://github.com/aloneguid/bt/releases/download/6.1.0/BrowserTamer-6.1.0-win64.msi):

<img width="310" height="94" alt="Image" src="https://github.com/user-attachments/assets/00edfdd0-bb1c-4563-a85f-cc8204b67e2d" />

<img width="339" height="351" alt="Image" src="https://github.com/user-attachments/assets/e12369f8-bb6d-4c11-8a23-f9ea6b4a8cb0" />

**Comment by aloneguid** [comment]
VT is at zero detections. Maybe update virus definitions? 

<img width="1881" height="1069" alt="Image" src="https://github.com/user-attachments/assets/be1ec5bf-449a-4de2-a3bf-2666487ee15c" />

**Comment by Abeatbowl** [comment]
Yep, it works after a definition update, sorry for the inconvenience and thanks for the quick answer.
